spec/jobs/user_email_spec.rb

```diff
@@ -288,6 +288,37 @@
       expect(mail.body).to include("asdfasdf")
     end
 
+    context "confirm_new_email" do
+      let(:email_token) { Fabricate(:email_token, user: user) }
+      before do
+        EmailChangeRequest.create!(
+          user: user,
+          requested_by: requested_by,
+          new_email_token: email_token,
+          new_email: "testnew@test.com",
+          change_state: EmailChangeRequest.states[:authorizing_new]
+        )
+      end
+
+      context "when the change was requested by admin" do
+        let(:requested_by) { Fabricate(:admin) }
+        it "passes along true for the requested_by_admin param which changes the wording in the email" do
+          Jobs::UserEmail.new.execute(type: :confirm_new_email, user_id: user.id, email_token: email_token.token)
+          mail = ActionMailer::Base.deliveries.first
+          expect(mail.body).to include("This email change was requested by a site admin.")
+        end
+      end
+
+      context "when the change was requested by the user" do
+        let(:requested_by) { user }
+        it "passes along false for the requested_by_admin param which changes the wording in the email" do
+          Jobs::UserEmail.new.execute(type: :confirm_new_email, user_id: user.id, email_token: email_token.token)
+          mail = ActionMailer::Base.deliveries.first
+          expect(mail.body).not_to include("This email change was requested by a site admin.")
```

Comment:
I think this spec is incorrect too.

Reply:
The change was requested by the user so `not_to include("This email change was requested by a site admin.")` is correct

Reply:
Ahh icic. I'm generally not a fan of `not_to include` because an incorrect body will also result in a passing test.

Reply:
@tgxworld I mean, the normal body has nothing special about it, and there are other tests to see if the token is inside the body. IMO not to include is fine for these special cases. The only other way I can test it is with the mocks, because the code that has the conditional is so deeply nested.

Reply:
Actually if another test is already testing for the right body, why do we need this test since the other tests would not have the `requested_by_admin` opts set to `true`. Another problem with depending on another test to test the right behavior is that if those tests change, the assumptions made in the current test may no longer hold true.

Reply:
I am not seeing this as a problem. This test is saying the body should not include "This email change was requested by a site admin." if the user requested their own email change. It is self contained; other tests test other things. I want to test that it does not include that text in this case, and that it does include the text in the other case. They are 2 different contexts/scenarios.